docs/core/deploying/native-aot/index.md

```diff
@@ -114,24 +122,48 @@ By default, native AOT publishing produces debug information in a separate file
 
 ## Limitations of native AOT deployment
 
-Native AOT applications come with a few fundamental limitations and compatibility issues. The key limitations include:
+### [.NET 7](#tab/net7)
+
+Native AOT apps have limitations and compatibility issues:
 
-- No dynamic loading (for example, `Assembly.LoadFile`).
-- No run-time code generation (for example, `System.Reflection.Emit`).
+- No dynamic loading, for example, `Assembly.LoadFile`.
+- No run-time code generation, for example, `System.Reflection.Emit`.
 - No C++/CLI.
-- No built-in COM (only applies to Windows).
+- Windows: No built-in COM.
 - Requires trimming, which has [limitations](../trimming/incompatibilities.md).
 - Implies compilation into a single file, which has known [incompatibilities](../single-file/overview.md#api-incompatibility).
 - Apps include required runtime libraries (just like [self-contained apps](../index.md#publish-self-contained), increasing their size as compared to framework-dependent apps).
 - <xref:System.Linq.Expressions> always use their interpreted form, which is slower than run-time generated compiled code.
+- Should be targeted for console type apps, ASP.NET Core is not supported.
+- Not all the runtime libraries are fully annotated to be native AOT compatible. That is, some warnings in the runtime libraries aren't actionable by end developers.
+- Limited diagnostic support for debugging and profiling.
 
-The publish process analyzes the entire project and its dependencies and produces warnings whenever the limitations of the published application might be encountered at run time.
+The publish process:
```

Comment:
I find this change problematic. Before this it reads like "one part of the publish process analyzes the entire project ...". After the change it reads like "the publish process is these two steps - analysis and producing warnings", which is wrong and misleading.

I think the original intent of this sentence was to point out that there are warning produced by the publish process which will alert the developer if most of the above limitations are encountered by the app. It was not here to describe what the publish process does (lot of things).

Reply:
Applies to the .net 8 version as well - same story.

Reply:
> I find this change problematic. Before this it reads like "one part of the publish process analyzes the entire project ...". After the change it reads like "the publish process is these two steps - analysis and producing warnings", which is wrong and misleading.
> 
> I think the original intent of this sentence was to point out that there are warning produced by the publish process which will alert the developer if most of the above limitations are encountered by the app. It was not here to describe what the publish process does (lot of things).

Can you suggest a rewriting of the original. The sentence is way too long and I was splitting it up.

Reply:
Maybe something like:
"The publish process will detect if running the app may encounter some of the limitations and will produce warnings accordingly". (not a native speaker, so my grammar is terrible)